### PR TITLE
Rewrite about_modules

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Rewrite about_modules

- Fixes [AB#190105](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/190105)

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md [hunk 1]
```diff
@@ -14,29 +14,27 @@ Describes  Windows PowerShell snap-ins and shows how to use and manage them.
...
-A Windows PowerShell snap-in is a Microsoft .NET Framework assembly that
-contains Windows PowerShell providers and\/or cmdlets. Windows PowerShell
-includes a set of basic snap-ins, but you can extend the power and value of
-Windows PowerShell by adding snap-ins that contain providers and cmdlets that
-you create or get from others.
+A Windows PowerShell snap-in is a Microsoft .NET Framework assembly that can
+contains Windows PowerShell providers and cmdlets. Windows PowerShell includes
+a set of basic snap-ins, but you can extend the power and value of Windows
+PowerShell by adding snap-ins that contain providers and cmdlets that you
+create or get from others.
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:21]: Should this be:

```suggestion
A Windows PowerShell snap-in is a Microsoft .NET Framework assembly that can
contain Windows PowerShell providers and cmdlets. Windows PowerShell includes a
set of basic snap-ins, but you can extend the power and value of Windows
PowerShell by adding snap-ins that contain providers and cmdlets that you
create or get from others.
```

or:

```suggestion
A Windows PowerShell snap-in is a Microsoft .NET Framework assembly that
contains Windows PowerShell providers and cmdlets. Windows PowerShell includes
a set of basic snap-ins, but you can extend the power and value of Windows
PowerShell by adding snap-ins that contain providers and cmdlets that you
create or get from others.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md [hunk 2]
```diff
@@ -14,29 +14,27 @@ Describes  Windows PowerShell snap-ins and shows how to use and manage them.
...
 When you add a snap-in, the cmdlets and providers that it contains are
 immediately available for use in the current session, but the change affects
 only the current session.
 
 To add the snap-in to all future sessions, save it in your Windows PowerShell
-profile. You can also use the Export-Console cmdlet to save the snap-in names
-to a console file and then use it in future sessions. You can even save
-multiple console files, each with a different set of snap-ins.
+profile. You can also use the `Export-Console` cmdlet to save the snap-in names
+to a console file and then use that saved configuration in future sessions.
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:29]: With the only remaining snap-in being Core, which is always loaded, is this information still relevant/useful? Or are we just keeping it for consistency, or are there still snap-ins we need to keep this information for?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md [hunk 3]
```diff
@@ -50,40 +48,29 @@ To get the snap-in for each  Windows PowerShell provider, type:
...
-```powershell
-Get-PSSnapin -registered
-```
-
-### ADDING THE SNAP-IN TO THE CURRENT SESSION
+## Add a snap-in to the current session
 
-To add a registered snap-in to the current session, use the Add-PsSnapin
+To add a registered snap-in to the current session, use the `Add-PsSnapin`
 cmdlet. For example, to add the Microsoft SQL Server snap-in to the session,
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:74]: 1. Do we use module formatting for snap-in names?
2. Is this snap-in still available and in use, or is it a module now?

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md [hunk 4]
```diff
@@ -92,49 +79,50 @@ Add-PSSnapin sql
...
 
-The providers and cmdlets in the snapin are now available for use in the session.
+The providers and cmdlets in the snap-in are now available for use in the
+session.
 
-### REMOVING A SNAP-IN
+## Remove a snap-in
 
 To remove a Windows PowerShell snap-in from the current session, use the
 Remove-PsSnapin cmdlet. For example, to remove the SQL Server snap-in from the
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:128]:
```suggestion
`Remove-PsSnapin` cmdlet. For example, to remove the SQL Server snap-in from the
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 5]
```diff
@@ -13,103 +13,61 @@ Explains how to install, import, and use PowerShell modules.
...
-modules to their PowerShell sessions and use them just like the built-in
-commands.
+- Snap-ins
 
-This topic explains how to use PowerShell modules. For information about how to
-write PowerShell modules, see
-[Writing a PowerShell Module](/powershell/scripting/developer/module/writing-a-windows-powershell-module).
+  Beginning in PowerShell 3.0 the Microsoft.PowerShell.Core snap-in is added to
+  every session by default. This is the only snap-in remaining in PowerShell.
+  All other snap-ins were converted to modules.
```
Comment (michaeltlombardi) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Modules.md:23]:
```suggestion
  Beginning in PowerShell 3.0 the **Microsoft.PowerShell.Core** snap-in is
  added to every session by default. This is the only snap-in remaining in
  PowerShell. All other snap-ins were converted to modules.
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 6]
```diff
@@ -13,109 +13,67 @@ Explains how to install, import, and use PowerShell modules.
 
 ## Long description
 
-A module is a package that contains PowerShell members, such as cmdlets,
-providers, functions, workflows, variables, and aliases.
+PowerShell provides two ways to package cmdlets, providers, functions,
+variables, and aliases:
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md:17]: This is covered in the class articles. Leave this out for now.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 7]
```diff
@@ -13,109 +13,67 @@ Explains how to install, import, and use PowerShell modules.
...
+many Windows features include modules for managing the feature. Those modules
+get installed when the feature is installed. Other modules may come in an
+installer or Setup program that installs the module.
 
 Use the following command to create a **Modules** directory for the current
 user:
 
 ```powershell
-New-Item -Type Directory -Path $HOME\Documents\WindowsPowerShell\Modules
+New-Item -Type Directory -Path $HOME\Documents\PowerShell\Modules
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md:76]: Need to add more context. Will rewrite this.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md [hunk 8]
```diff
@@ -125,283 +83,251 @@ MyModule folder from `C:\ps-test\MyModule` to the Modules directory, type:
...
-To remove a module from your session, use the following command format.
-
-```powershell
-Remove-Module <module-name>
-```
-
-For example, the following command removes the BitsTransfer module from the
-current session.
+the session. For example, the following command removes the **BitsTransfer**
+module from the current session.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Modules.md:241]: Nope

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:29]: None we care about.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_PSSnapins.md:74]: I don't think it matters for this context.

**Review by michaeltlombardi — APPROVED** [review]

**Pull request merged**